includes/plugins/co-authors-plus/class-guest-contributor-role.php

```diff
@@ -75,6 +74,47 @@ public static function init() {
 		add_action( 'wp_update_user', [ __CLASS__, 'edit_user_profile_update' ] );
 	}
 
+	/**
+	 * Runs early in the init hook to make sure it runs before Co-Authors Plus initialization.
+	 *
+	 * @return void
+	 */
+	public static function early_init() {
+		if ( defined( 'NEWSPACK_ENABLE_CAP_GUEST_AUTHORS' ) && NEWSPACK_ENABLE_CAP_GUEST_AUTHORS ) {
+			return;
+		}
+		if ( ! self::site_has_cap_guest_authors() ) {
+			add_filter( 'coauthors_guest_authors_enabled', '__return_false' );
+			add_action( 'admin_menu', [ __CLASS__, 'guest_author_menu_replacement' ] );
+		}
+	}
+
+	/**
+	 * Checks if the site has any guest authors. Will check it once in the database and store the result in an option.
```

Comment:
I wonder if the once-ever check is not too restrictive. Without it (or with a periodical check, until there are no GAs), the migration path is simple: delete GAs and profit. The added step of having to delete the option necessitates the involvement of a TAM or developer.

Reply:
hmm good point. I'll add a wp_cron to check it periodically, I just don't want doing this query on every request

Reply:
@adekbadek Added the cron check and updated test instructions.

I also added a condition to refresh the check every time a guest author is deleted, so we would immediately update the UI, but the experience is not great. When you delete the last Guest Author, WP still redirects you to the Guest Authors page, but since we disable GAs, you end up in an error page because that page is not registered...

I removed it, but if you have an idea on how to make it better, here's what's needed:

```
add_action( 'delete_post', [ __CLASS__, 'guest_author_deleted' ], 10, 2 );

public static function guest_author_deleted( $post_id, $post ) {
	if ( 'guest-author' === $post->post_type ) {
		self::clear_site_has_cap_guest_authors_check();
	}
}
```